test/spec/Feature/Query/PgSafeUpdateSpec.hs

```diff
@@ -74,101 +66,56 @@ spec =
-            , { "id": 2, "name": "updated-item", "observation": null }
-            , { "id": 3, "name": "updated-item", "observation": null }
-            ]|]
-
-        request methodPost "/rpc/reset_items_tables"
-          [("Prefer", "tx=commit")]
-          [json| {"tbl_name": "unsafe_update_items"} |]
-          `shouldRespondWith` ""
-          { matchStatus  = 204 }
+      it "works if no condition is present" $
+        verifyMutation "unsafe_update_items" tblDataBefore
+          [json|[
+            { "id": 1, "name": "updated-item", "observation": null }
+          , { "id": 2, "name": "updated-item", "observation": null }
+          , { "id": 3, "name": "updated-item", "observation": null }
+          ]|]
+          $
+          request methodPatch "/unsafe_update_items"
+              [("Prefer", "tx=commit"), ("Prefer", "count=exact")]
+              [json| {"name": "updated-item"} |]
+            `shouldRespondWith`
+              ""
+              { matchStatus  = 204
+              , matchHeaders = [ matchHeaderAbsent hContentType
+                               , "Content-Range" <:> "0-2/3"
+                               , "Preference-Applied" <:> "tx=commit" ]
+              }
 
     context "Full table delete" $ do
-      it "works if no condition is present" $ do
-        get "/unsafe_delete_items"
-          `shouldRespondWith`
-            [json|[
-              { "id": 1, "name": "item-1", "observation": null }
-            , { "id": 2, "name": "item-2", "observation": null }
-            , { "id": 3, "name": "item-3", "observation": null }
-            ]|]
-
-        request methodDelete "/unsafe_delete_items"
-            [("Prefer", "tx=commit"), ("Prefer", "count=exact")]
-            mempty
-          `shouldRespondWith`
-            ""
-            { matchStatus  = 204
-            , matchHeaders = [ matchHeaderAbsent hContentType
-                             , "Content-Range" <:> "*/3"
-                             , "Preference-Applied" <:> "tx=commit" ]
-            }
-
-        get "/unsafe_delete_items?order=id"
-          `shouldRespondWith`
-            [json|[]|]
-
-        request methodPost "/rpc/reset_items_tables"
-          [("Prefer", "tx=commit")]
-          [json| {"tbl_name": "unsafe_delete_items"} |]
-          `shouldRespondWith` ""
-          { matchStatus  = 204 }
+      it "works if no condition is present" $
+        verifyMutation "unsafe_delete_items" tblDataBefore
+          [json|[]|]
+          $
+          request methodDelete "/unsafe_delete_items"
+              [("Prefer", "tx=commit"), ("Prefer", "count=exact")]
+              mempty
+            `shouldRespondWith`
+              ""
+              { matchStatus  = 204
+              , matchHeaders = [ matchHeaderAbsent hContentType
+                               , "Content-Range" <:> "*/3"
+                               , "Preference-Applied" <:> "tx=commit" ]
+              }
```

Comment:
We're always specifying the `tx=commit` plus checking the `Prefernce-Applied`, could this be abstracted away?

I also note that in every test the `shouldRespondWith` is comparing to `""`.  Maybe we can ignore the status/headers and just focus on the result body? This should make the tests even shorter.